x/ecocredit/basket/msg_create.go

```diff
@@ -0,0 +1,50 @@
+package basket
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+var (
+	_ legacytx.LegacyMsg = &MsgCreate{}
+)
+
+///
+/// MsgCreate sdk.Msg interface
+///
+
+const nameMaxLen = 32
+
+// ValidateBasic does a stateless sanity check on the provided data.
+func (m MsgCreate) ValidateBasic() error {
```

Comment:
left to check: 

- [ ] timestamp nil?
- [ ] valid fee?
- [ ] credit type given?
- [ ] allowed classes >0 ?

Reply:
let's confirm 1 & 2 in chat.